### PR TITLE
NAS-117239 / 22.12 / NAS-117239: 2 new endpoints for webUI network changes

### PR DESCRIPTION
**Pull request by AlexKarpov98**
**Testing:**
Network => Interfaces =>Add
When the form is submitted you should show a dialog to register a default gateway.
Screenshot_45

**_Note:_** An error is caused from the endpoint when a default gateway is saved.
Screenshot_46

**Bug report:**
This error is caused because the int_address and int_netmask are empty in an interface created.
I confirmed it calling `ws.call('datastore.query', 'network.interfaces')` endpoint.
Screenshot_44

It's necessary for the middleware team to check this, I think.

### CODE REVIEW

File: src/app/pages/network/components/default-gateway-dialog/default-gateway-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public cdr: ChangeDetectorRef,
+    private dialogRef: MatDialogRef<DefaultGatewayDialogComponent>,
+    private errorHandler: FormErrorHandlerService,
+  ) {}
+
+  onSubmit(): void {
+    this.dialogRef.close();
+    const formValues = this.form.value;
+    this.ws.call('interface.save_default_route', [formValues.defaultGateway]).pipe(untilDestroyed(this)).subscribe(
+      () => {
```
Comment (denysbutenko) [inline src/app/pages/network/components/default-gateway-dialog/default-gateway-dialog.component.ts:42]: Nit: You can pass object with error key only

File: src/app/pages/network/components/interface-form/interface-form.component.ts [hunk 2]
```diff
@@ -217,6 +222,15 @@ export class InterfaceFormComponent implements OnInit {
         this.isLoading = false;
         this.core.emit({ name: 'NetworkInterfacesChanged', data: { commit: false, checkin: false }, sender: this });
         this.slideInService.close();
+
+        this.ws.call('interface.default_route_will_be_removed').pipe(untilDestroyed(this)).subscribe((res) => {
```
Comment (denysbutenko) [inline src/app/pages/network/components/interface-form/interface-form.component.ts:226]: filter(Boolean) to avoid “res” and nesting

**Review by denysbutenko — COMMENTED** [review]
I still need to test it

**Review by denysbutenko — CHANGES_REQUESTED** [review]
Validation errors do not appear:
image
image

Perhaps this change is blocked until the middleware issue is resolved
image

File: src/app/pages/network/components/default-gateway-dialog/default-gateway-dialog.component.ts (hunk 1 above)
Comment (AlexKarpov98) [reply src/app/pages/network/components/default-gateway-dialog/default-gateway-dialog.component.ts]: Fixed

File: src/app/pages/network/components/interface-form/interface-form.component.ts (hunk 2 above)
Comment (AlexKarpov98) [reply src/app/pages/network/components/interface-form/interface-form.component.ts]: Thanks. Fixed it

**Comment by AlexKarpov98** [comment]
> Validation errors do not appear: image image
> 
> Perhaps this change is blocked until the middleware issue is resolved image

Fixed.

**Review by denysbutenko — APPROVED** [review]
Looks good

**Pull request merged**